File: 1-easy/problems/0404-sum-of-left-leaves/solution.py
def sumOfLeftLeaves(root):
    def dfs(node, is_left):
        if not node:
            return 0

        if not node.left and not node.right and is_left:
            return node.val

        return dfs(node.left, True) + dfs(node.right, False)

    return dfs(root, False)

File: 1-easy/problems/0404-sum-of-left-leaves/solution.ts
function sumOfLeftLeaves(root: TreeNode | null): number {
    function dfs(node: TreeNode | null, isLeft: boolean): number {
        if (!node) return 0;

        if (!node.left && !node.right && isLeft) {
            return node.val;
        }

        return dfs(node.left, true) + dfs(node.right, false);
    }

    return dfs(root, false);
}
